CMakeLists.txt

```diff
@@ -94,6 +94,7 @@ if(UNIX AND NOT APPLE) # i.e.: Linux
     option(BUILD_WSI_XCB_SUPPORT "Build XCB WSI support" ON)
     option(BUILD_WSI_XLIB_SUPPORT "Build Xlib WSI support" ON)
     option(BUILD_WSI_WAYLAND_SUPPORT "Build Wayland WSI support" ON)
+    option(BUILD_WSI_HEADLESS_SUPPORT "Build Headless WSI support" ON)
```

Comment:
Yep, it is intended to be cross platform. I'll make the change to have it be enabled all of the time.

As a related question, the loader itself will never report support for a particular surface type right? At least not on its own. In order for a surface type to reported during an vkEnumerateInstanceExtensionProperties call an underlying icd or layer has to export support for it?

For instance VK_KHR_wayland_surface is listed in loader_extension_generator.py in the WSI_EXT_NAMES list, but that list is used to remove reporting of those extensions as supported by the loader. Feel free to educate me here, but I think what the loader does is it needs to support allocating the instance level object (VkIcdSurface) for a particular surface type, and these are listed in the OutputInstantExtensionWhitelistArray function in loader_extension_generator.py.. If an extension is reported by a layer or icd, and the loader via that white list, then it gets reported. If it's not supported by the loader, then nope. If it's supported by the loader but not any icd or layer, then nope.

The reason I'm asking is because I'm thinking we should be following the same pattern for the headless surface type..an underlying layer or icd should be reporting support for it for it to be exposed instead of the loader always returning that it's supported.

Reply:
> As a related question, the loader itself will never report support for a particular surface type right? At least not on its own. In order for a surface type to reported during an vkEnumerateInstanceExtensionProperties call an underlying icd or layer has to export support for it?

That is all correct. So I think you should be covered here.